File: UI/rbike_admin/lib/models/order_item.dart
import 'equipment.dart';

class OrderItem {
  final int? orderItemsId;
  final int orderId;
  final int equipmentId;
  final int quantity;
  final Equipment? equipment;

  OrderItem({
    this.orderItemsId,
    required this.orderId,
    required this.equipmentId,
    required this.quantity,
    this.equipment,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      orderItemsId: json['orderItemsId'],
      orderId: json['orderId'],
      equipmentId: json['equipmentId'],
      quantity: json['quantity'],
      equipment: json['equipment'] != null 
          ? Equipment.fromJson(json['equipment']) 
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'orderItemsId': orderItemsId,
      'orderId': orderId,
      'equipmentId': equipmentId,
      'quantity': quantity,
    };
  }
} 

File: rBike.Model/OrderItem.cs
using System;
using System.Collections.Generic;

namespace rBike.Model
{
    public class OrderItem
    {
        public int OrderItemsId { get; set; }

        public int OrderId { get; set; }

        public int EquipmentId { get; set; }

        public int Quantity { get; set; }

        public virtual Equipment Equipment { get; set; } = null!;

        public virtual Order Order { get; set; } = null!;
    }
}
